src/main/java/htsjdk/tribble/gff/Gff3Codec.java

```diff
@@ -0,0 +1,432 @@
+                        }
+
+                        // check for strand
+
+                        canDecode &= fields[GENOMIC_STRAND_INDEX].equals(Strand.POSITIVE.toString()) ||
+                                fields[GENOMIC_STRAND_INDEX].equals(Strand.NEGATIVE.toString()) ||
+                                fields[GENOMIC_STRAND_INDEX].equals(Strand.NONE.toString()) ||
+                                fields[GENOMIC_STRAND_INDEX].equals("?");
+                    }
+                }
+
+            }
+        }
+        catch (FileNotFoundException ex) {
+            logger.error(inputFilePath + " not found.");
+            return false;
+        }
+        catch (final IOException ex) {
+            return false;
+        }
+
+        return canDecode;
+    }
+
+    @Override
+    public FeatureCodecHeader readHeader(LineIterator lineIterator) {
+
+        List<String> header = new ArrayList<>();
+        while(lineIterator.hasNext()) {
+            String line = lineIterator.peek();
+            if (line.startsWith(COMMENT_START)) {
+                header.add(line);
+                lineIterator.next();
+            } else {
+                break;
+            }
+        }
+
+        return new FeatureCodecHeader(header, FeatureCodecHeader.NO_HEADER_END);
+    }
+
+    /**
+     * Parse a directive line from a gff3 file
+     * @param directiveLine
+     * @throws IOException
+     */
+    private void parseDirective(final String directiveLine) throws IOException {
+        final Gff3Directive directive = Gff3Directive.toDirective(directiveLine);
+        if (directive != null) {
+            processDirective(directive, directive.decode(directiveLine));
+
+        } else {
+            logger.warn("ignoring directive " + directiveLine);
+        }
+    }
+
+    /**
+     * Process a gff3 directive
+     * @param directive the gff3 directive, indicated by a specific directive line
+     * @param decodedResult the decoding of the directive line by the given directive
+     */
+    private void processDirective(final Gff3Directive directive, final Object decodedResult) {
+        switch (directive) {
+            case VERSION3_DIRECTIVE:
+                break;
+
+            case SEQUENCE_REGION_DIRECTIVE:
+                final SequenceRegion newRegion = (SequenceRegion) decodedResult;
+                if (sequenceRegionMap.containsKey(newRegion.getContig())) {
+                    throw new TribbleException("directive for sequence-region " + newRegion.getContig() + " included more than once.");
+                }
+                sequenceRegionMap.put(newRegion.getContig(), newRegion);
+                break;
+
+            case FLUSH_DIRECTIVE:
+                prepareToFlushFeatures();
+                break;
+
+            case FASTA_DIRECTIVE:
+                reachedFasta = true;
+                break;
+
+            default:
+                throw new IllegalArgumentException( "directive " + directive + " not handled by parser.");
```

Comment:
This exception is there to catch if a constant is added to the `Gff3Directive`  enum which then is not handled by this switch.  So I think `IllegalArgumentException` is more appropriate, since this exception is intended to catch the future introduction of a bug, not an issue with corrupt input data.  Perhaps an `AssertionError` or `IllegalStateException` would be better, my internet searches where inconclusive on the best practice for this type of situation.   I will improve the error message to make more explicit what's going on here.